src/partial.js

```diff
@@ -0,0 +1,23 @@
+import fetch from 'isomorphic-fetch'
+import normalize from './utilities/normalize'
+
+/*
+  defines a partial request,
+  returns a new function that merges any values
+*/
+
+export default (strings, ...vars) => {
+  let { options, url } = normalize(strings, vars)
+
+  return (strings, ...vars) => {
+    let finalData = normalize(strings, vars)
+    let mergedOptions = { ...options, ...finalData.options }
+
+    return new Promise((resolve, reject) => {
```

Comment:
may want to split this actual fetching part into it's own place. We will have to change this here and in request.js when we add middleware